File: Country2City.php
<?php

/*
	Author: 			  Fatih Mert Doğancan
	Github:				  fatihmert
	Release Data:		01.06.2017 - 06:15

	Note:				    When you searching anything please write latin abc (not allow çşıüğö) 
*/


class Country2City{
	private $file, $arr;
	public function __construct(){
		$this->file = file_get_contents("country2city.json");
		$this->arr = json_decode($this->file,true);
		return $this;
	}

	public function getCities($countryName){
		if(isset($this->arr[(string)$countryName])){
			return $this->arr[(string)$countryName];
		}else{
			return 0;
		}
	}

	public function getCountries(){
		return array_keys($this->arr);
	}

	public function findCountryFromCity($cityName){
		foreach ($this->arr as $key => $value) {
			foreach ($value as $city) {
				if((string)$city == (string)$cityName){
					return $key;
				}
			}
		}
		return 0;
	}
}

/*
  Example Using
*/

echo "<pre>;
$c2c = new Country2City();

echo "<h1>Cities of Turkey</h1><br/>";
print_r($c2c->getCities("Turkey"));

echo "Istanbul in ".$c2c->findCountryFromCity("Istanbul");

echo "<br/><h1>Countries of World</h1><br/>";
print_r($c2c->getCountries());

?>

File: Country2City.py
#-*- coding: utf-8 -*-

# Author: 			  	Fatih Mert Doğancan
# Github:				fatihmert
# Release Data:			15.06.2017 - 03:27
# Note:				    When you searching anything please write latin abc (not allow çşıüğö) 

import json

class Country2City(object):
	def __init__(self):
		self.file = open("country2city.json")
		self.arr = json.load(self.file)
		return self

	def isset(self,exp):
		try:
		    exp
		    return 1;
		except IndexError:
		    return 0;

	def getCities(countryName):
		if self.isset(self.arr[str(countryName)]):
			return self.arr[str(countryName)]
		else:
			return 0


	def getCountries():
		return self.arr.keys()

	def findCountryFromCity(cityName):
		for k,v in self.arr:
			for city in v:
				if(str(city) == str(cityName)):
					return k
		return 0

# Example Using

c2c = Country2City()

c2c.getCities("Turkey")
print "Istanbul in ".c2c.findCountryFromCity("Istanbul")
c2c.getCountries()
